torch/ao/quantization/observer.py

```diff
@@ -406,68 +478,26 @@ def __init__(
         quant_min=None,
         quant_max=None,
         factory_kwargs=None,
-        memoryless=False,
-    ) -> None:
-
-        # For x86 quantized kernels, we need to ensure that the vpmaddubsw
-        # instruction does not overflow. We allow for a reduce_range argument to
-        # observers that reduces the quantized range to (0,127) or (-64, 63).
-        # For more details see aten/src/ATen/native/quantized/cpu/qconv.cpp
-        # This is not an optimal choice for non x86 backends as it loses a bit
-        # of precision for activations.
+        compute_dtype=None,
+    ):
+        # memoryless should not be a user facing argument, infer it from compute dtype and dtype
+        memoryless = compute_dtype in (torch.qint8,
+                                       torch.quint8,
+                                       torch.quint4x2,
+                                       torch.qint32,) and dtype == torch.float32
         super(MinMaxObserver, self).__init__(
-            dtype=dtype,
+            dtype=dtype if compute_dtype is None else compute_dtype,
```

Comment:
maybe we can always pass in compute_dtype, but set compute_dtype to dtype when compute_dtype is None

Reply:
ya, I made this change, it still feels a little weird to set compute_dtype to dtype and then set dtype to compute_dtype when we pass it in, but I'm not sure if there's a better way.